Essentials/src/main/java/com/earth2me/essentials/commands/Commandessentials.java

```diff
@@ -577,52 +575,77 @@ private void runHomes(final Server server, final CommandSource sender, final Str
         }
     }
 
-    // Forces a rerun of userdata UUID conversion.
-    private void runUUIDConvert(final Server server, final CommandSource sender, final String commandLabel, final String[] args) throws Exception {
-        sender.sendMessage("Starting Essentials UUID userdata conversion; this may lag the server.");
-
-        final Boolean ignoreUFCache = args.length > 2 && args[1].toLowerCase(Locale.ENGLISH).contains("ignore");
-        EssentialsUpgrade.uuidFileConvert(ess, ignoreUFCache);
-
-        sender.sendMessage("UUID conversion complete. Check your server log for more information.");
-    }
-
-    // Looks up various UUIDs for a user.
-    private void runUUIDTest(final Server server, final CommandSource sender, final String commandLabel, final String[] args) throws Exception {
-        if (args.length < 2) {
-            throw new Exception("/<command> uuidtest <name>");
+    // Gets information about cached users
+    private void runUserMap(final CommandSource sender, final String[] args) {
+        if (!sender.isAuthorized("essentials.usermap", ess)) {
+            return;
         }
-        final String name = args[1];
-        sender.sendMessage("Looking up UUID for " + name);
 
-        UUID onlineUUID = null;
+        final ModernUserMap userMap = ess.getUsers();
+        sender.sendMessage(tl("usermapSize", userMap.getCachedCount(), userMap.getUserCount(), ess.getSettings().getMaxUserCacheCount()));
+        if (args.length > 1) {
+            if (args[1].equals("full")) {
+                for (final Map.Entry<String, UUID> entry : userMap.getNameCache().entrySet()) {
+                    sender.sendMessage(tl("usermapEntry", entry.getKey(), entry.getValue().toString()));
+                }
+            } else if (args[1].equals("purge")) {
+                final boolean seppuku = args.length > 2 && args[2].equals("iknowwhatimdoing");
 
-        for (final Player player : ess.getOnlinePlayers()) {
-            if (player.getName().equalsIgnoreCase(name)) {
-                onlineUUID = player.getUniqueId();
-                break;
-            }
-        }
+                sender.sendMessage(tl("usermapPurge", String.valueOf(seppuku)));
 
-        final UUID essUUID = ess.getUserMap().getUser(name).getConfigUUID();
+                final Set<UUID> uuids = new HashSet<>(ess.getUsers().getAllUserUUIDs());
+                ess.runTaskAsynchronously(() -> {
+                    final File userdataFolder = new File(ess.getDataFolder(), "userdata");
+                    final File backupFolder = new File(ess.getDataFolder(), "userdata-npc-backup-boogaloo");
```

Comment:
Include the date/time in the folder name? Would make it easier for admins to identify if they did it multiple times